app/views/guests/show.html.erb

```diff
@@ -0,0 +1,13 @@
+<h1>Guest Name: <%= @guest.name %></h1>
+<% @guest.rooms.each do |room| %>
+  <h3>Room Number(s): </h3><%= room.id %>
+  <h3>Room Rate: </h3><%= room.rate %>
+  <h3>Room Suite: </h3><%= room.suite %>
+  <h3>Hotel Name: </h3><%=  room.hotel.name %>
+<% end %>
+
+<%= form_with url: "/guests/#{@guest.id}/add_room/room_id", method: :post, local:true do |form| %>
```

Comment:
I believe the problem you were getting was your params in the Guests controller for this form had "room_id" => "room_id", instead of the number of the id you were putting into the form. The reason for this is on this line, your form_with url has the word "room_id" hard coded in it, so when the route see this, it sets the dynamic part of the url `:room_id` (from your routes.rb file) to this value of "room_id". See my comment in the routes about the making these restful, because this route does not follow restful conventions for a post. We can't know the id of the roomuntil the user submits it.

Reply:
It was close though!